File: DecompiledGameScripts/core/gameplay/prereqs/damage/targetTypeHitPrereqCondition.swift

public class TargetTypeHitPrereqCondition extends BaseHitPrereqCondition {

  public let m_targetType: CName;

  public func SetData(recordID: TweakDBID) -> Void {
    this.m_targetType = TweakDBInterface.GetCName(recordID + t".targetType", n"None");
    super.SetData(recordID);
  }

  public func Evaluate(hitEvent: ref<gameHitEvent>) -> Bool {
    let result: Bool;
    switch this.m_targetType {
      case n"Player":
        result = IsDefined(hitEvent.target as PlayerPuppet);
        break;
      case n"Puppet":
        result = IsDefined(hitEvent.target as ScriptedPuppet);
        break;
      case n"Vehicle":
        result = IsDefined(hitEvent.target as VehicleObject);
        break;
      default:
        return false;
    };
    if result {
      result = this.CheckOnlyOncePerShot(hitEvent);
    };
    return this.m_invert ? !result : result;
  }
}

File: Definitions/Game_Definitions/classes/TargetTypeHitPrereqCondition.lua
---@meta
---@diagnostic disable

---@class TargetTypeHitPrereqCondition : BaseHitPrereqCondition
---@field targetType CName
TargetTypeHitPrereqCondition = {}

---@return TargetTypeHitPrereqCondition
function TargetTypeHitPrereqCondition.new() return end

---@param props table
---@return TargetTypeHitPrereqCondition
function TargetTypeHitPrereqCondition.new(props) return end

---@param hitEvent gameeventsHitEvent
---@return Bool
function TargetTypeHitPrereqCondition:Evaluate(hitEvent) return end

---@param recordID TweakDBID|string
function TargetTypeHitPrereqCondition:SetData(recordID) return end

